vertx-web-openapi-router/src/main/java/io/vertx/ext/web/openapi/router/impl/SecuritySchemeImpl.java

```diff
@@ -0,0 +1,193 @@
+        }
+
+        if (!"digest".equals(config.getString("scheme"))) {
+          throw new IllegalStateException("Invalid 'schema' value for HTTP security scheme: " + config.getString("scheme"));
+        }
+
+        routerBuilder.security(securitySchemeId, handler, null);
+        return routerBuilder;
+      }
+    }
+    throw new IllegalStateException("OpenAPI does not contain securityScheme: " + securitySchemeId);
+  }
+
+  @Override
+  public RouterBuilder httpHandler(JWTAuthHandler handler) {
+    Objects.requireNonNull(handler, "'handler' cannot be null");
+    JsonObject securitySchemes = contract.getRawContract().getJsonObject("components").getJsonObject("securitySchemes");
+
+    if (securitySchemes != null) {
+      if (securitySchemes.containsKey(securitySchemeId)) {
+        final JsonObject config = securitySchemes.getJsonObject(securitySchemeId);
+
+        if (!"http".equals(config.getString("type"))) {
+          throw new IllegalStateException("Invalid 'type' value for HTTP security scheme: " + config.getString("type"));
+        }
+
+        if (!"bearer".equals(config.getString("scheme"))) {
+          throw new IllegalStateException("Invalid 'schema' value for HTTP security scheme: " + config.getString("scheme"));
+        }
+
+        routerBuilder.security(securitySchemeId, handler, null);
+        return routerBuilder;
+      }
+    }
+    throw new IllegalStateException("OpenAPI does not contain securityScheme: " + securitySchemeId);
+  }
+
+  @Override
+  public RouterBuilder oauth2Handler(String callback, Function<JsonObject, OAuth2AuthHandler> factory) {
+    Objects.requireNonNull(factory, "'factory' cannot be null");
+    JsonObject securitySchemes = contract.getRawContract().getJsonObject("components").getJsonObject("securitySchemes");
+
+    if (securitySchemes != null) {
+      if (securitySchemes.containsKey(securitySchemeId)) {
+        final JsonObject config = securitySchemes.getJsonObject(securitySchemeId);
+
+        if (!"oauth2".equals(config.getString("type"))) {
+          throw new IllegalStateException("Invalid 'type' value for Oauth2 security scheme: " + config.getString("type"));
+        }
+
+        if (config.containsKey("authorizationCode")) {
+          // callback is required for authorizationCode flow
+          Objects.requireNonNull(callback, "'callback' cannot be null when using authorizationCode flow");
+        }
+
+        routerBuilder.security(securitySchemeId, factory.apply(config), callback);
+        return routerBuilder;
+      }
+    }
+    throw new IllegalStateException("OpenAPI does not contain securityScheme: " + securitySchemeId);
+  }
+
+
+  @Override
+  public Future<RouterBuilder> openIdConnectHandler(String callback, Function<String, Future<OAuth2AuthHandler>> factory) {
+    JsonObject securitySchemes = contract.getRawContract().getJsonObject("components").getJsonObject("securitySchemes");
```

Comment:
I'd like to see this in the OpenAPIContract. Then we can check already there if all required config entries are available.

Reply:
Fixed by using the https://github.com/eclipse-vertx/vertx-openapi/pull/21 api